app/services/middleware_topology_service.rb

```diff
@@ -33,14 +36,14 @@ def entity_display_type(entity)
 
   def build_entity_data(entity)
     data = build_base_entity_data(entity)
-    data.merge!(:status => 'Unknown',
-                :display_kind => entity_display_type(entity))
-    data[:icon] = entity.decorate.try(:listicon_image)
+    data[:status] = 'Unknown'
+    data[:display_kind] = entity_display_type(entity)
+    data[:icon] = entity.decorate.try(:listicon_image) unless entity.kind_of?(MiddlewareDatasource)
```

Comment:
perhaps worth here to do it slightly differently: try to get the font icon from decorator, if empty - get the image from decorator. that can also solve adding things to ui mixin, because currently the icon is defined twice - once in the decorator and once in uimixin